### PR TITLE
Add support for AJP connections

### LINKED ISSUE #88: AJP support in 3.0+
We should retain the support of AJP13 in 3.0+ for loadbalancing purposes.

http://www.appsdev.is.ed.ac.uk/blog/?p=525

### PR DESCRIPTION
**Pull request by tkrille**
Resolves #88

### CODE REVIEW

**Comment by tkrille** [comment]
@timopick Can you please verify, whether this works as you expect?

**Comment by wallner** [comment]
This all looks good to me, however I don't really see the point of it, since it's not possible to cluster  OSIAM at the moment.

**Comment by timopick** [comment]
@wallner We're using AJP connectors in a current project- The cluster ability should however be restored at some point in the future.

**Comment by wallner** [comment]
@timopick I agree with you: OSIAM needs to be clusterable, however it never has been, but we have it on the road map.

**Comment by timopick** [comment]
Looks good! Thanks!

**Comment by dacrome** [comment]
So you closed this PR, @timopick?

**Comment by timopick** [comment]
Lol, oops on my part :) I meant to close the feature request ;)

**Comment by tkrille** [comment]
@timopick If this is the desired solution, please merge the PR. I've already assigned you to this.

File: src/main/java/org/osiam/configuration/AjpConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    @Value("${osiam.ajp.port:8009}")
+    private int port;
+
+    @Override
+    public void customize(ConfigurableEmbeddedServletContainer container) {
+        if (container instanceof TomcatEmbeddedServletContainerFactory) {
+            TomcatEmbeddedServletContainerFactory tomcat = (TomcatEmbeddedServletContainerFactory) container;
+            Connector ajpConnector = new Connector("AJP/1.3");
+            ajpConnector.setProtocol("AJP/1.3");
```
Comment (dacrome) [inline src/main/java/org/osiam/configuration/AjpConfiguration.java]: Already set via constructor

File: src/main/java/org/osiam/configuration/AjpConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            ajpConnector.setAttribute("address", bindAddress);
+            ajpConnector.setPort(port);
+            ajpConnector.setSecure(false);
+            ajpConnector.setAllowTrace(false);
+            ajpConnector.setScheme("http");
```
Comment (dacrome) [inline src/main/java/org/osiam/configuration/AjpConfiguration.java:53]: I'm not sure, but shouldn't be all configurable?
- protocol
- secure
- allowTrace
- scheme

**Comment by dacrome** [comment]
Please rebase

**Comment by tkrille** [comment]
rebased

File: src/main/java/org/osiam/configuration/AjpConfiguration.java (hunk 2 above)
Comment (tkrille) [inline src/main/java/org/osiam/configuration/AjpConfiguration.java:53]: I don't know. This is something for @timopick to decide. I've just implemented the most minimal solution. We can still add more configuration properties later.

File: src/main/java/org/osiam/configuration/AjpConfiguration.java (hunk 1 above)
Comment (tkrille) [inline src/main/java/org/osiam/configuration/AjpConfiguration.java]: Yes, changed it.

File: src/main/java/org/osiam/configuration/AjpConfiguration.java (hunk 2 above)
Comment (dacrome) [inline src/main/java/org/osiam/configuration/AjpConfiguration.java:53]: Ok

**Comment by dacrome** [comment]
Thank you!

**Pull request merged**